solidity/contracts/BancorUpgradeConverter.sol

```diff
@@ -0,0 +1,236 @@
+
+        @return the new converter
+    */
+    function createConverter(IBancorConverter _fromConverter) private returns(IBancorConverter) {
+        IERC20Token emptyConnector = IERC20Token(address(0));
+        ISmartToken token = _fromConverter.token();
+        IBancorConverterExtensions extensions = _fromConverter.extensions();
+        uint32 maxConversionFee = _fromConverter.maxConversionFee();
+        BancorConverter converter = new BancorConverter(
+            token,
+            extensions,
+            maxConversionFee,
+            emptyConnector,
+            0
+        );
+        return IBancorConverter(converter);
+    }
+
+    /**
+        @dev copies the connectors from the current converter to the new one
+
+        @param _fromConverter       current converter to copy from
+        @param _toConverter         new converter to add connectors
+        @param _formerVersions      true if the converter version is under 0.5
+    */
+    function copyConnectors(IBancorConverter _fromConverter, IBancorConverter _toConverter, bool _formerVersions)
+        private
+    {
+        uint256 virtualBalance;
+        uint32 weight;
+        bool isVirtualBalanceEnabled;
+        bool isPurchaseEnabled;
+        bool isSet;
+        for (uint16 i = 1; i < _fromConverter.convertibleTokenCount(); i++) {
+            address connectorAddress = _fromConverter.convertibleToken(i);
+            (virtualBalance, weight, isVirtualBalanceEnabled, isPurchaseEnabled, isSet) = readConnector(
+                _fromConverter,
+                connectorAddress,
+                _formerVersions
+            );
+            ISmartToken connectorToken = ISmartToken(connectorAddress);
+            _toConverter.addConnector(connectorToken, weight, isVirtualBalanceEnabled);
+        }
+    }
+
+    /**
+        @dev copies the conversion fee from the current converter to the new one
+
+        @param _fromConverter       current converter to copy from
+        @param _toConverter         new converter to add a conversion fee
+    */
+    function copyConvertionFee(IBancorConverter _fromConverter, IBancorConverter _toConverter)
+        private
+    {
+        uint32 conversionFee = _fromConverter.conversionFee();
+        _toConverter.setConversionFee(conversionFee);
+    }
+
+    /**
+        @dev copies the quick buy path array from the current converter to the new one
+
+        @param _fromConverter       current converter to copy from
+        @param _toConverter         new converter to add a path
+    */
+    function copyQuickBuyPath(IBancorConverter _fromConverter, IBancorConverter _toConverter)
+        private
+    {
+        uint256 quickBuyPathLength = _fromConverter.getQuickBuyPathLength();
+        if (quickBuyPathLength > 0) {
+            IERC20Token[] memory path = new IERC20Token[](quickBuyPathLength);
+            for (uint256 i = 0; i < quickBuyPathLength; i++) {
+                path[i] = _fromConverter.quickBuyPath(i);
```

Comment:
might be more efficient to create a local var for the quick buy path and iterate over it instead of calling another contract's function in each iteration

Reply:
we can call to specific item every time, we can't get all array in a single call